### PR TITLE
Remove 'component' span attribute in instrumentations

### LINKED ISSUE #300: Remove 'component' span attribute in instrumentations
The semantic convention for the `component` attritubte was removed in the specfication some time ago. See https://github.com/open-telemetry/opentelemetry-specification/issues/336

There still exist some instrumentations (e.g. requests, aiohttp-client, everything using dbapi, ...) which add this attriubte.
Since it was removed in the specification instrumentations shouldn't set this attribute anymore.

### PR DESCRIPTION
**Pull request by mariojonke**
# Description

Fixes #300 

## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated

### CODE REVIEW

**Review by lzchen — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py [hunk 1]
```diff
@@ -107,14 +107,15 @@ def wrap_connect(
...
 
     Args:
-        tracer: The :class:`opentelemetry.trace.Tracer` to use.
         connect_module: Module name where connect method is available.
         connect_method_name: The connect method name.
-        database_component: Database driver name or database name "JDBI",
-            "jdbc", "odbc", "postgreSQL".
+        database_system: An identifier for the database management system (DBMS)
+            product being used.
         database_type: The Database type. For any SQL database, "sql".
```
Comment (codeboten) [inline instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py:114]: should database_type be removed? i couldn't find it in the semantic conventions for databases

**Review by codeboten — COMMENTED** [review]
PR looks great, just one question regarding whether database_type should be removed as well.

**Review by owais — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py (hunk 1 above)
Comment (mariojonke) [reply instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py:114]: Yes looks like there are no semantic convetions for it. It also doesn't seem to be used anywhere in the code so i'll remove it.

**Review by toumorokoshi — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Thanks for addressing my comments!

**Pull request merged**